subprojects/language-native/src/main/java/org/gradle/language/cpp/plugins/CppLibraryPlugin.java

```diff
@@ -172,26 +175,51 @@ public void execute(MavenPublication publication) {
                             }
                         });
-                        extension.getPublications().create("debug", MavenPublication.class, new Action<MavenPublication>() {
+                        extension.getPublications().create("debugShared", MavenPublication.class, new Action<MavenPublication>() {
                             @Override
                             public void execute(MavenPublication publication) {
                                 // TODO - should track changes to these properties
                                 publication.setGroupId(project.getGroup().toString());
-                                publication.setArtifactId(library.getBaseName().get() + "_debug");
+                                publication.setArtifactId(library.getBaseName().get() + "_debugShared");
                                 publication.setVersion(project.getVersion().toString());
-                                NativeVariant debugVariant = new NativeVariant("debug", linkUsage, debugLinkElements, runtimeUsage, debugRuntimeElements);
+                                NativeVariant debugVariant = new NativeVariant("debugShared", linkUsage, debugLinkElements, runtimeUsage, debugRuntimeElements);
                                 mainVariant.addVariant(debugVariant);
                                 publication.from(debugVariant);
                             }
                         });
-                        extension.getPublications().create("release", MavenPublication.class, new Action<MavenPublication>() {
+                        extension.getPublications().create("releaseShared", MavenPublication.class, new Action<MavenPublication>() {
                             @Override
                             public void execute(MavenPublication publication) {
                                 // TODO - should track changes to these properties
                                 publication.setGroupId(project.getGroup().toString());
-                                publication.setArtifactId(library.getBaseName().get() + "_release");
+                                publication.setArtifactId(library.getBaseName().get() + "_releaseShared");
                                 publication.setVersion(project.getVersion().toString());
-                                NativeVariant releaseVariant = new NativeVariant("release", linkUsage, releaseLinkElements, runtimeUsage, releaseRuntimeElements);
+                                NativeVariant releaseVariant = new NativeVariant("releaseShared", linkUsage, releaseLinkElements, runtimeUsage, releaseRuntimeElements);
+                                mainVariant.addVariant(releaseVariant);
+                                publication.from(releaseVariant);
+                            }
+                        });
+
+                        extension.getPublications().create("debugStatic", MavenPublication.class, new Action<MavenPublication>() {
```

Comment:
I think we just want a debug module with all of the debug binaries in it, and a release module with all of the release binaries in it.